src/main/java/uk/gov/hmcts/ccd/DefaultCacheWarmUpService.java

```diff
@@ -0,0 +1,63 @@
+package uk.gov.hmcts.ccd;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.http.HttpHeaders;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.ccd.data.definition.DefaultCaseDefinitionRepository;
+import uk.gov.hmcts.ccd.idam.AuthenticatedUser;
+import uk.gov.hmcts.ccd.idam.IdamHelper;
+import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator;
+
+import java.util.List;
+
+@Service
+@Qualifier(DefaultCacheWarmUpService.QUALIFIER)
+public class DefaultCacheWarmUpService implements CacheWarmUpService {
+
+    private static final Logger LOG = LoggerFactory.getLogger(CachingConfiguration.class);
+    public static final String QUALIFIER = "default";
+    private static final long TWO_SECONDS = 2000l;
+
+    private final DefaultCaseDefinitionRepository caseDefinitionRepository;
+    private final AuthTokenGenerator authTokenGenerator;
+    private final ApplicationParams applicationParams;
+    private final IdamHelper idamHelper;
+
+    public DefaultCacheWarmUpService(final DefaultCaseDefinitionRepository caseDefinitionRepository,
+                                     final AuthTokenGenerator authTokenGenerator,
+                                     final ApplicationParams applicationParams,
+                                     final IdamHelper idamHelper) {
+        this.caseDefinitionRepository = caseDefinitionRepository;
+        this.authTokenGenerator = authTokenGenerator;
+        this.applicationParams = applicationParams;
+        this.idamHelper = idamHelper;
+    }
+
+    @Override
+    public void warmUp() {
+        try {
+            HttpHeaders httpHeaders = authorizationHeaders();
+            List<String> caseTypesReferences = caseDefinitionRepository.getCaseTypesReferences(httpHeaders);
+            for (String reference : caseTypesReferences) {
+                Thread.sleep(TWO_SECONDS);
```

Comment:
I'm always a bit concerned on using directly Thread sleep. Can we see if there's any more modern approach to execute operations with sleep time in between? perhaps some library, spring support, etc...

Reply:
Just come to my mind that we will probably need this Pre Warming to be an asynchronous activity. As it's now, if it takes 1 min to load all the case type definition,
we will be slowing down Data Store startup time by 1 min?

Reply:
Yes I will look at the async way.

Reply:
👍